packages/flutter_tools/test/general.shard/dap/flutter_adapter_test.dart

```diff
@@ -25,10 +25,9 @@ void main() {
           : '/fake/flutter';
     });
 
-
-  group('launchRequest', () {
-    test('runs "flutter run" with --machine', () async {
-      final MockFlutterDebugAdapter adapter = MockFlutterDebugAdapter(fileSystem: globals.fs, platform: globals.platform);
+    group('launchRequest', () {
+      test('runs "flutter run" with --machine', () async {
+        final MockFlutterDebugAdapter adapter = MockFlutterDebugAdapter(fileSystem: globals.fs, platform: globals.platform);
```

Comment:
They don't use them, these ones are just simple unit tests. I've changed them to use `MemoryFileSystem.test()` and `FakePlatform`, but I made the platform match the real one so that the Windows bots will test with Windows paths etc.

(I haven't verified on Windows so if I didn't do this right the bots may fail, I'll check back on this later since I gotta be away for a little)

Reply:
This does fail on windows:

```
00:48 +1256 ~3 -2: test\general.shard\dap\flutter_test_adapter_test.dart: flutter test adapter includes toolArgs [E]                                                                                   
  Expected: 'C:\\fake\\flutter\\bin\\flutter.bat'
    Actual: 'C:\\fake\\flutter/bin/flutter.bat'
     Which: is different.
            Expected: ... e\\flutter\\bin\\flu ...
              Actual: ... e\\flutter/bin/flutt ...
                                    ^
             Differ at offset 17
  
  package:test_api                                            expect
  test\general.shard\dap\flutter_test_adapter_test.dart 49:7  main.<fn>.<fn>
```

I think when constructing the MemoryFileSystem, you have to also specify the FileSystemStyle in the constructor